pallets/task/src/lib.rs

```diff
@@ -695,15 +694,14 @@ pub mod pallet {
 
 			// remove task from storage
 			<Tasks<T>>::remove(task_id);
-			// Transfer balance amount from escrow account to task creator
-			let sub_account = Self::account_id(&task_id);
-			<T as self::Config>::Currency::transfer(&sub_account, &task_initiator, task.budget,
-				ExistenceRequirement::AllowDeath)?;
+			
+			// Unreserve balance amount from task creator
+			<T as self::Config>::Currency::unreserve(&task_initiator, task.budget);
```

Comment:
I believe this is correct as is because the extrinsic it is used in is remove_task() not accept_task(). 

Therefore the task creator has not marked as completed is just removing the task alltogether. Would it be better to not allow the deletion of tasks if someone has been assigned?

Reply:
This is the state machine for the tasks: https://github.com/UniversalDot/documents/blob/master/designs/architecture/ProcessDesign.drawio.png 

Deletion is allowed only if its in a state Created. See line 693. 

Any improvements we can do to the state machine?

Reply:
I think the state machine looks good. My mistake, i seemed to miss the ensure statement!